test/suite/test_prepare02.py

```diff
@@ -0,0 +1,123 @@
+#!/usr/bin/env python
+#
+# Public Domain 2014-2018 MongoDB, Inc.
+# Public Domain 2008-2014 WiredTiger, Inc.
+#
+# This is free and unencumbered software released into the public domain.
+#
+# Anyone is free to copy, modify, publish, use, compile, sell, or
+# distribute this software, either in source code form or as a compiled
+# binary, for any purpose, commercial or non-commercial, and by any
+# means.
+#
+# In jurisdictions that recognize copyright laws, the author or authors
+# of this software dedicate any and all copyright interest in the
+# software to the public domain. We make this dedication for the benefit
+# of the public at large and to the detriment of our heirs and
+# successors. We intend this dedication to be an overt act of
+# relinquishment in perpetuity of all present and future rights to this
+# software under copyright law.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+# IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+# test_prepare02.py
+#   Prepare : check post conditions to prepare operation
+#
+
+from suite_subprocess import suite_subprocess
+import wiredtiger, wttest
+
+def timestamp_str(t):
+    return '%x' % t
+
+class test_prepare02(wttest.WiredTigerTestCase, suite_subprocess):
+    def test_prepare_session_operations(self):
+        if not wiredtiger.timestamp_build():
+            self.skipTest('requires a timestamp build')
+
+        self.session.create("table:mytable", "key_format=S,value_format=S")
+        cursor = self.session.open_cursor("table:mytable", None)
+        # Session operations not permitted after prepare_transaction
+        self.session.begin_transaction()
+        self.session.prepare_transaction()
+        msg = "/ not permitted in a/"
+        # Below operations are not supported in prepared state. Operations
+        # are listed in the same order as declared with in session
+        # structure and assigned in open_session. Any function missed below
+        # is to be considered as supported in prepare transaction state.
```

Comment:
I recommend a bit of a rewording:
 \#
 \# The operations listed below are not supported in the prepared state.
 \#
 \# The operations are listed in the same order as they are declared in the session
 \# structure. Any function missing below is allowed in the prepared state.
 \#